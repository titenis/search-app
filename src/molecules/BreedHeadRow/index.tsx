import TableCell from '@material-ui/core/TableCell';
import { SortLabel } from 'molecules/SortLabel';
import TableRow from '@material-ui/core/TableRow';
import React, { FC } from 'react';

export const BreedHeadRow: FC = () => (
  <TableRow>
    <TableCell>
      <SortLabel name="name">Name</SortLabel>
    </TableCell>
    <TableCell align="right">
      <SortLabel name="origin">Origin</SortLabel>
    </TableCell>
    <TableCell align="right">Life span</TableCell>
    <TableCell align="right">
      <SortLabel name="temperament">Temperament</SortLabel>
    </TableCell>
    <TableCell align="right">Weight (kg)</TableCell>
    <TableCell align="right">Wiki</TableCell>
  </TableRow>
);
