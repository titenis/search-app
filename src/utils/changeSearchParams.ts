export const changeSearchParams = (
  params: URLSearchParams,
  key: string,
  value?: string,
): URLSearchParams => {
  if (value) {
    params.set(key, value);
  } else {
    params.delete(key);
  }

  return params;
};
