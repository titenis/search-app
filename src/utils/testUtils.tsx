import React, { FC, ReactElement } from 'react';

import { render, RenderResult } from '@testing-library/react';
import { createMemoryHistory, History } from 'history';
import { Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { theme } from 'theme';
import { QueryClient, QueryClientProvider } from 'react-query';
import { renderHook, RenderHookResult } from '@testing-library/react-hooks';

export const Providers = (history: History): FC => ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        cacheTime: 0,
      },
    },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <Router history={history}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </Router>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  { history = createMemoryHistory(), ...restOptions } = {},
): RenderResult =>
  render(ui, {
    wrapper: Providers(history),
    ...restOptions,
  });

export const customRenderHook = <P, R>(
  hook: (props?: P) => R,
  { history = createMemoryHistory(), ...options } = {},
): RenderHookResult<P, R> =>
  renderHook(hook, {
    wrapper: Providers(history),
    ...options,
  });

export * from '@testing-library/react';

export { customRender as render };
export { customRenderHook as renderHook };
