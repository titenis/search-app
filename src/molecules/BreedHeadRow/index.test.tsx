import React from 'react';
import { render } from 'utils/testUtils';
import { BreedHeadRow } from 'molecules/BreedHeadRow/index';

describe('BreedHeadRow', () => {
  it('should render', () => {
    const { container } = render(
      <table>
        <thead>
          <BreedHeadRow />
        </thead>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });
});
