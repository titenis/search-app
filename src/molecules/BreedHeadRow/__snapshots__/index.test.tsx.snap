// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedHeadRow should render 1`] = `
<div>
  <table>
    <thead>
      <tr
        class="MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiTableSortLabel-root"
            role="button"
            tabindex="0"
          >
            Name
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiTableSortLabel-root"
            role="button"
            tabindex="0"
          >
            Origin
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          Life span
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiTableSortLabel-root"
            role="button"
            tabindex="0"
          >
            Temperament
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          Weight (kg)
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignRight"
        >
          Wiki
        </td>
      </tr>
    </thead>
  </table>
</div>
`;
